### ISSUE TITLE
Keyboard accessibility navigation not working on MonthYearPicker

### ISSUE THREAD

**Issue by Orhan92** [Hacker0x01/react-datepicker#4400]
**Describe the bug**
It is not possible to navigate with keyboard navigation in the YearMonthPicker. When I click on the input field there is no way to use keyboard navigation to navigate through the calendar. This is not considered accessible and there is no way for the user to navigate to the calendar without first clicking on it. It works like a charm in the standard calendar with dates but not in MonthYearPicker.

**To Reproduce**
1. Use prop: showMonthYearPicker
2. Try to navigate with keyboard

**Expected behavior**
When calendar pops up the user should be able to navigate through the calendar with keyboard. In other words, when the input field is clicked on or selected, keyboard navigation should set focus on the calendar.

**Desktop (please complete the following information):**

- OS: Windows 10/11
- Browser: Chrome:latest / Firefox:latest

**Additional context**
Is there any solution to this applied in the repo or is it actually a bug?

**Comment by lasseklovstad** [comment]
Added a pr to hopefully fix this. Temporary fix i'm using to allow up/down arrow key focus first month:
````
<DatePicker
onKeyDown={(e) => {
          if (!(e.target instanceof HTMLInputElement)) {
            return;
          }
          e.preventDefault();
          const eventKey = e.key;
          const selector = ".react-datepicker__month-text[tabindex='0']";
          if (eventKey === "ArrowDown" || eventKey === "ArrowUp") {
            const selectedItem =
              document.querySelector<HTMLDivElement>(selector);
            selectedItem && selectedItem.focus();
          }
        }} />
```
